### PR TITLE
feat(tenkz): add corpus render baselines

### LINKED ISSUE #26: Adopt an in-repo regression corpus for the tensor-network library
The full test corpus (~235 files: smoke/stress/gallery/boundary/cups/atoms/plane/sheets tests, the twelve RMP hard-figure reproductions, edge-case probes) lives in a session-local scratchpad and dies with the session. The repo carries only the small acceptance set in tex/tenkz/examples/. Nothing session-local can be the regression suite that 0.9/1.0 claim.

Tasks:
- [ ] Curate the corpus into tests/tenkz/ (or tex/tenkz/examples/ extended): every file compiles standalone, states its formula in % comments, and has a one-line header naming what it regression-tests
- [ ] RMP hard-figure reproductions included as .tex only; the paper originals are referenced by arXiv figure number in comments, not committed
- [ ] A driver script (scripts/tenkz_corpus.sh): compiles every file with the canonical invocation, fails on any error, reports a count; optional --render mode producing PNGs for visual review
- [ ] scripts/tenkz_audit.py runs over every produced .tnlog; hard findings fail the driver
- [ ] CI: the corpus job added to .github/workflows/pr-ci.yml, triggered on tex/tenkz/** changes
- [ ] Baseline discipline documented: behavior-changing PRs regenerate affected renders and say so; the LionSR/TNLean#4158 gate relies on this harness for its "renders unchanged" rule

Canonical invocation (from the corpus directory):
  timeout 120 env TEXINPUTS="$REPO/tex/tenkz//:" xelatex -interaction=nonstopmode -halt-on-error <file>.tex

### PR DESCRIPTION
**Pull request by LionSR**
## Summary

- add an opt-in `scripts/tenkz_corpus.sh --render` mode for the adopted 257-fixture corpus
- render every PDF page at 200 dpi into deterministic per-fixture PNG paths
- write stable `CENSUS.tsv` and `SHA256SUMS` manifests for before/after review
- protect custom output directories with a non-symlink regular ownership marker and staged installation
- document the LionSR/TNLean#4158 baseline discipline and route the focused safety test through CI

## Safety contract

- arbitrary nonempty, symlinked, protected, or overlapping output paths are rejected
- all rendering and manifest construction finish before the live baseline is touched
- a pre-commit installation failure restores the untouched previous baseline
- after the complete new tree is live, backup cleanup is best effort; failure keeps the new baseline authoritative and reports the residual backup path

## Validation

- `python3 scripts/test_tenkz_corpus_render.py`
- `python3 scripts/test_tenkz_corpus_provenance.py`
- `TENKZ_CORPUS_VALIDATE_ONLY=1 scripts/tenkz_corpus.sh` (257 fixtures)
- two full real `scripts/tenkz_corpus.sh --render` runs: 257 PDFs, 259 pages, identical census and checksum manifests
- all 259 generated checksums verified; representative 16-page contact sheet inspected
- ShellCheck, `bash -n`, `py_compile`, actionlint, reader-prose check, `git diff --check`
- tenkz source lint: 116 files, 0 findings

Closes LionSR/tenkz#26

---

> [!NOTE]
> **Low Risk**
> Changes are confined to optional corpus rendering tooling, CI path filters, and documentation; default `tenkz_corpus.sh` behavior is unchanged without `--render`.
> 
> **Overview**
> Adds an **opt-in `--render` path** on `scripts/tenkz_corpus.sh` so the adopted `tests/tenkz` corpus can produce a **repository-wide 200 dpi PNG baseline** (default `build/tenkz-corpus-render/`) after the usual compile-and-audit pass, with `--render-dir` resolved from the repo root.
> 
> New **`scripts/tenkz_render_corpus.py`** turns compiled corpus PDFs into per-fixture `page-NNN.png` trees plus **`CENSUS.tsv`** and **`SHA256SUMS`**, using staged install, destination locking, ownership markers, and protected paths so failed or concurrent runs do not clobber unrelated data or the last good baseline.
> 
> **`scripts/test_tenkz_corpus_render.py`** exercises that safety contract and driver CLI; **PR CI** runs it in the `tenkz-corpus` job and watches the new scripts in path filters. Docs in **`tests/tenkz/README.md`**, **`docs/tenkz/HACKING.md`**, and **`docs/tenkz/DESIGN.md`** describe the before/after checksum workflow for issue **#4158** behavior-preserving changes.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 4f08ffe1e44dd7d78d00ef8b96d536ab20eca926. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: scripts/tenkz_render_corpus.py [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+    try:
+        marker_status = marker.lstat()
+    except OSError:
+        return False
+    if not stat.S_ISREG(marker_status.st_mode):
+        return False
+    try:
+        return marker.read_text(encoding="utf-8") == MARKER_CONTENT
+    except OSError:
+        return False
```
[the remark this answers is not in the document]
Comment (LionSR) [reply scripts/tenkz_render_corpus.py:46]: Fixed on exact head d879186b5: marker decoding now catches (OSError, UnicodeError). Added a malformed UTF-8 regular-marker regression that asserts clean refusal, no traceback, and no user-data mutation. python3 scripts/test_tenkz_corpus_render.py passes.

File: scripts/tenkz_render_corpus.py [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+
+def install_stage(stage: Path, output_dir: Path) -> None:
+    backup = output_dir.parent / f".{output_dir.name}.previous.{os.getpid()}"
+    if backup.exists():
+        raise RenderError(f"stale render backup blocks installation: {backup}")
+
+    moved_old = output_dir.exists()
+    if moved_old:
+        try:
+            output_dir.rename(backup)
```
[the remark this answers is not in the document]
Comment (LionSR) [reply scripts/tenkz_render_corpus.py:271]: Fixed on exact head 4f08ffe1e. Installers now take a nonblocking advisory lock keyed to the destination, record the pre-render destination inode/tree snapshot, compare it again immediately before installation, and verify the captured tree after the atomic move before any backup can be deleted. A changed destination is restored and the install is refused. Added a deterministic regression that starts from an empty destination, injects unowned user data after rendering begins, and verifies explicit refusal with the data unchanged. python3 scripts/test_tenkz_corpus_render.py and the focused provenance/257-corpus/static gates pass.

File: scripts/tenkz_render_corpus.py [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+    ]
+    (stage / "SHA256SUMS").write_text("\n".join(checksum_lines) + "\n", encoding="utf-8")
+
+
+def install_stage(stage: Path, output_dir: Path) -> None:
+    backup = output_dir.parent / f".{output_dir.name}.previous.{os.getpid()}"
+    if backup.exists():
+        raise RenderError(f"stale render backup blocks installation: {backup}")
+
+    moved_old = output_dir.exists()
```
[the remark this answers is not in the document]
Comment (LionSR) [reply scripts/tenkz_render_corpus.py:268]: Addressed on exact head 4f08ffe1e. The destination is now revalidated under a destination-specific advisory lock immediately before install, with pre-render and post-atomic-move inode/tree snapshot comparisons guarding the narrower race. If the directory was populated or otherwise changed, installation is refused and captured data is restored rather than deleted. The deterministic empty-directory race regression preserves injected user data and passes.
